geo_autoRIFT/autoRIFT/autoRIFT.py

```diff
@@ -325,15 +319,37 @@ def autorift(self):
 
         Flag = 3
 
+        print(self.ChipSize0X)
+        print(ChipSizeUniX)
+
+        # define course filter
+        DispFiltC = DISP_FILT()
+        #            if ChipSizeUniX[i] == ChipSizeUniX[0]: #ASG ----------------
+        DispFiltC.FracValid = self.FracValid*self.GridSize_ChipSize0_Ratio#ASG ----------------
+        DispFiltC.FracSearch = self.FracSearch #ASG ----------------
+        DispFiltC.FiltWidth = int(((self.FiltWidth+1)*self.GridSize_ChipSize0_Ratio) - 1) #ASG ----------------
+        DispFiltC.Iter = int(self.Iter -1) #ASG ----------------
+        DispFiltC.MadScalar = int(self.MadScalar) #ASG ----------------
+
+        # define fine filter
+        DispFiltF = DISP_FILT()
+        #            if ChipSizeUniX[i] == ChipSizeUniX[0]: #ASG
+        DispFiltF.FracValid = self.FracValid
+        DispFiltF.FracSearch = self.FracSearch
+        DispFiltF.FiltWidth = self.FiltWidth
+        DispFiltF.Iter = self.Iter
+        DispFiltF.MadScalar = self.MadScalar
+
+
         for i in range(ChipSizeUniX.__len__()):
-            
+
             # Nested grid setup: chip size being ChipSize0X no need to resize, otherwise has to resize the arrays
-            if self.ChipSize0X != ChipSizeUniX[i]:
-                Scale = self.ChipSize0X / ChipSizeUniX[i]
+            if self.ChipSize0X != (ChipSizeUniX[i]*self.GridSize_ChipSize0_Ratio): #ASG
```

Comment:
chipsize0x -> grid_spacing -> from the file (e.g. 120 m)
GridSize_ChipSize0_Ratio -> GridSpacing_ChipSize0_Ratio -> GridSpacing_ChipSize0_oversample_ratio -> define chipsize0 and calculate the ratio

Reply:
I think the condition check can be the same as before regardless of whhat gridSpacing is. No need to change it. See below please.
**OLD**:
```
if self.ChipSize0X != ChipSizeUniX[i]:
        Scale = self.ChipSize0X / ChipSizeUniX[i]
```
**NEW** (what we discussed yesterday):
```
if self.GridSpacingX != (ChipSizeUniX[i] * GS_CS_oversample_ratio):
        Scale = self.GridSpacingX / (ChipSizeUniX[i] * GS_CS_oversample_ratio)
```
If you plug in `GridSpacing=120`, `ChipSize0=240`,  `ChipSizeUniX=[240, 480]`  (`GS_CS_oversample_ratio=0.5`), you will find the **NEW** one is equivalent to the **OLD** one.
However, the **NEW** one sometimes causes problems, while the **OLD** doesn’t. For example, when `GridSpacing=480`, `ChipSize0=240`,  `ChipSizeUniX=[240, 480]`  (`GS_CS_oversample_ratio=1`), the condition will be invalid when `ChipSizeUniX[i]=480` , hhowever we do want it to be valid.
For this reason, I think it is conveninent to keep using the **OLD** condition check.